lib/rucio/transfertool/fts3.py

```diff
@@ -321,20 +321,55 @@ def build_job_params(
     """
     Prepare the job parameters which will be passed to FTS transfertool
+    Please refer to https://fts3-docs.web.cern.ch/fts3-docs/fts-rest/docs/bulk.html#parameters
+    for the list of parameters.
     """
 
     # The last hop is the main request (the one which triggered the whole transfer),
     # so most attributes will come from it
     last_hop = transfer_path[-1]
     first_hop = transfer_path[0]
 
-    overwrite, bring_online_local = True, None
+    # Overwriting by default is set to True for non TAPE RSEs.
+    # Tape RSEs can force overwrite by setting the "overwrite" attribute to True.
+    # There is yet another configuration option: transfers->overwrite_corrupted_files that when is set to True
+    # it will retry failed requests with overwrite flag set to True
+    overwrite, overwrite_when_only_on_disk, bring_online_local = True, False, None
     if first_hop.src.rse.is_tape_or_staging_required():
         # Activate bring_online if it was requested by first hop
         # We don't allow multihop via a tape, so bring_online should not be set on any other hop
         bring_online_local = bring_online
     if last_hop.dst.rse.is_tape():
-        overwrite = False
+        # FTS v3.12.12 introduced a new boolean parameter "overwrite_when_only_on_disk" that controls if the file can be overwritten
+        # in TAPE enabled RSEs ONLY IF the file is on the disk buffer and not yet commited to tape media.
+        # This functionality should reduce the number of stuck files in the disk buffer that are not migrated to tape media (for whatever reason).
+        # Please be aware that FTS does not guarantee an atomic operation from the time it checks for existence of the file on disk and tape and
+        # the moment the file is overwritten, so there is a race condition that could overwrite the file on the tape media
+        overwrite = last_hop.dst.rse.attributes.get('overwrite', False)  # honour RSE configuration to force overwrite for all requests
+        overwrite_when_only_on_disk = last_hop.dst.rse.attributes.get('overwrite_when_only_on_disk', False)
+
+        # setting both flags is incompatible, so we opt in for the safest approach: "overwrite_when_only_on_disk"
+        # this is aligned with FTS implementation: see
+        if overwrite and overwrite_when_only_on_disk:
+            overwrite = False
+
+    if not (overwrite and overwrite_when_only_on_disk):
+        # if there are no overwrite flags set, we still need to check for the
+        # "transfers -> overwrite_corrupted_files setting. The logic behind this flag is that
+        # it will update the rws (RequestWithSources) with the "overwrite" attribute set to True
+        # after finding an 'Destination file exists and overwrite is not enabled' error message
+        overwrite_corrupted_files = last_hop.rws.attributes.get('overwrite', False)
+        if overwrite_corrupted_files:
+            overwrite = True  # both for DISK and TAPE
```

Comment:
It's only considered for tape.
We always overwrite for Disks.

Reply:
Thanks!